mcstatus/pinger.py

```diff
@@ -34,91 +31,94 @@ def handshake(self) -> None:
 
         self.connection.write_buffer(packet)
 
-    def read_status(self) -> JavaStatusResponse:
-        request = Connection()
-        request.write_varint(0)  # Request status
-        self.connection.write_buffer(request)
+    @abstractmethod
+    def read_status(self) -> JavaStatusResponse | Awaitable[JavaStatusResponse]:
+        """Make a status request and parse the response."""
+        raise NotImplementedError
```

Comment:
Raise is not necessary, as you cannot call an abstract method

Reply:
You kinda can though.
![image](https://github.com/user-attachments/assets/8444e9d5-408f-4e38-9a8b-f6889764a1d8)

Also, the same is done in `BaseStatusResponse` abc in `responses.py` for the abstract build method, so it's not like this is introducing in a new pattern to handle abstractmethods into the repo.

Reply:
Unless it's a static method... Sure, this was just a note

Reply:
Even if the abstract method is not static, it can be called. Python has zero safety for this. You could be thinking of a compiled language such as C++ or Java where abstract virtual methods that have no code cannot be compiled and thus result in a compilation error.

Reply:
I mean, *as long as you are not actively trying to call an abstract method*, you can't call an abstract method